framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -843,17 +865,40 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
         eventData->IsNull() ? nullptr : eventData->AllocateOutputData(1, { XR_TYPE_EVENT_DATA_BUFFER, nullptr });
     InitializeOutputStructNext(eventData);
 
-    XrResult replay_result;
+    // If we have not previously started tracking the event behavior for this event type, start tracking it.
+    if (event_behavior_tracking_.find(capture_event->type) == event_behavior_tracking_.end())
+    {
+        event_behavior_tracking_[capture_event->type] = EventBehaviorTracking();
+    }
 
-    // WIP: Put this constant somewhere interesting
-    const uint32_t kRetryLimit      = 10;
-    const int64_t  kMaxSleepLimitNs = 500000000; // 500ms
-    uint32_t       retry_count      = 0;
+    XrResult replay_result;
+    bool     assert_on_difference = true;
 
     if (out_eventData && capture_event)
     {
-        int64_t sleep_time = 1;
-        do
+        uint32_t retry_limit = 24;
+        uint32_t retry_count = 0;
+        int64_t  sleep_time  = 1;
+
+        if (event_behavior_tracking_[capture_event->type].allow_timeout_reduction)
+        {
+            // Adjust the time we wait so that we don't wait as long by dropping the retry count
+            // by the number of times we've missed the event previously.
+            uint32_t current_reduction = event_behavior_tracking_[capture_event->type].current_timeout_reduction;
+            if (current_reduction >= retry_limit)
+            {
+                retry_limit = 0;
```

Comment:
Needs to be one, or we won't even look for the event once.  We're not going to drop anything, as we save anything `other` that we get in `previously_received_unhandled_events_`  (unless we think there's going to be a storm of missing events)